### PR TITLE
Kamila/add fragment viewmodel

### PR DESCRIPTION
**Pull request by kamilababayeva**
Add view model to store view and context in fragment

### CODE REVIEW

File: app/src/main/java/ch/sdp/vibester/helper/ViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package ch.sdp.vibester.helper
+
+import android.content.Context
+import android.view.View
+
+/*
```
Comment (MaximeZmt) [inline app/src/main/java/ch/sdp/vibester/helper/ViewModel.kt:6]:
```suggestion
/**
```

File: app/src/main/java/ch/sdp/vibester/fragment/SearchUserFragment.kt [hunk 2]
```diff
@@ -49,22 +50,23 @@ class SearchUserFragment : Fragment(R.layout.fragment_search_user), OnItemClickL
...
-    private fun setScanBtnListener(view: View) {
-        val buttonScan: FloatingActionButton = view.findViewById(R.id.searchUser_scanning)
+    private fun setScanBtnListener() {
+        val buttonScan: FloatingActionButton = vmSearchUser.view.findViewById(R.id.searchUser_scanning)
 
         buttonScan.setOnClickListener {
-            val qrIntent = Intent(requireActivity(), QrScanningActivity::class.java)
+            val qrIntent = Intent(activity, QrScanningActivity::class.java)
             qrIntent.putExtra("uidList", uidList)
             startActivity(qrIntent)
```
Comment (MaximeZmt) [inline app/src/main/java/ch/sdp/vibester/fragment/SearchUserFragment.kt:71]: I know you didn't wrote that part (only modify it)
But If you have time can you refactor it with the intent switcher.
Also if doable test it.

File: app/src/main/java/ch/sdp/vibester/fragment/ScoreBoardFragment.kt [hunk 3]
```diff
@@ -111,24 +100,24 @@ class ScoreBoardFragment : Fragment(), OnItemClickListener {
...
     /**
      * @param position index of the item on click
      * go to the profile of the player at index position
      */
     override fun onItemClick(position: Int) {
-        val intent = Intent(requireActivity(), PublicProfileActivity::class.java)
+        val intent = Intent(activity, PublicProfileActivity::class.java)
         intent.putExtra("UserId", players[position].uid)
         intent.putExtra("ScoresOrFollowing", R.string.profile_scores.toString() )
         startActivity(intent)
```
Comment (MaximeZmt) [inline app/src/main/java/ch/sdp/vibester/fragment/ScoreBoardFragment.kt:116]: Also if possible and you have time, can you refactor it with intent Switcher ?

**Review by MaximeZmt — APPROVED** [review]
LGTM 👍  Great Job ! Hope it will improve stability!
There is one minor typo that you can correct and two refactor that you may do if you have some time.

File: app/src/main/java/ch/sdp/vibester/fragment/GameSetupFragment.kt [hunk 4]
```diff
@@ -36,77 +34,83 @@ import retrofit2.Response
...
-        view.findViewById<Button>(R.id.validateSearch).setOnClickListener(this)
-
-        setReturnBtnListener(view)
-        setSpinnerListener(view, ctx, R.id.difficulty_spinner, R.array.difficulties_name)
-        setSpinnerListener(view, ctx, R.id.size_spinner, R.array.game_size_options)
+        vmGameSetup.view = view
+        vmGameSetup.ctx = view.context
+
+        setGenreListeners()
+        setGameModeListeners()
```
Comment (Tsathogguaa) [inline app/src/main/java/ch/sdp/vibester/fragment/GameSetupFragment.kt:53]: Good way to refactor this! Nice job!

File: app/src/main/java/ch/sdp/vibester/fragment/SearchUserFragment.kt [hunk 5]
```diff
@@ -113,7 +115,7 @@ class SearchUserFragment : Fragment(R.layout.fragment_search_user), OnItemClickL
     }
 
     override fun onItemClick(position: Int) {
-        val intent = Intent(requireActivity(), PublicProfileActivity::class.java)
+        val intent = Intent(activity, PublicProfileActivity::class.java)
         intent.putExtra("UserId", users[position].uid)
         intent.putExtra("ScoresOrFollowing", R.string.profile_following.toString())
         startActivity(intent)
```
Comment (Tsathogguaa) [inline app/src/main/java/ch/sdp/vibester/fragment/SearchUserFragment.kt:121]: Same thing as what Maxime had said, could you look into refactoring this with the Intent Switcher?

**Review by Tsathogguaa — APPROVED** [review]
Good work overall! Just the intentSwitcher stuff that can be changed around, but it's not as severe to complete ASAP. If you have time, would be great to adapt them!

**Pull request merged**